server/routes/login.js

```diff
@@ -1,207 +1,33 @@
 const { NotifyClient } = require('notifications-node-client')
 const ipRangeCheck = require('ip-range-check')
-const asyncMiddleware = require('../middleware/asyncMiddleware')
+const jwtDecode = require('jwt-decode')
 const logError = require('../logError')
 const config = require('../../config')
-const health = require('../controllers/health')
-const log = require('../../log')
 const utilities = require('../helpers/utilities')
-const staffMemberService = require('../services/staffMemberService')
 const userAuthenticationService = require('../services/userAuthenticationService')
 
-const {
-  notify: { url, clientKey, smsTemplateId, emailTemplateId },
-} = config
-const notify = url ? new NotifyClient(url, clientKey) : new NotifyClient(clientKey)
-
-const notifySmsTemplate = smsTemplateId || ''
-const notifyEmailTemplate = emailTemplateId || ''
-
 module.exports = () => (router) => {
-  router.get(
-    '/',
-    asyncMiddleware(async (req, res) => {
-      postLogin(req, res)
-    }),
-  )
-
-  router.get(
-    '/auth/login',
-    asyncMiddleware(async (req, res) => {
-      postLogin(req, res)
-    }),
-  )
-
-  router.post(
-    '/auth/2fa',
-    asyncMiddleware(async (req, res) => {
-      const userAuthenticationDetails = await userAuthenticationService.getUserAuthenticationDetails(req.user.username)
-
-      const inputTwoFactorCode = utilities.createTwoFactorAuthenticationHash(req.body.code)
-
-      if (inputTwoFactorCode === userAuthenticationDetails[0].TwoFactorAuthenticationHash) {
-        req.user.employeeName = await getStaffMemberEmployeeName(
-          userAuthenticationDetails[0].ApiUrl,
-          utilities.getStartMonth(),
-          req.user.token,
-        )
-
-        await userAuthenticationService.updateUserSessionExpiryAndLastLoginDateTime(
-          req.user.username,
-          new Date(Date.now() + config.hmppsCookie.expiryMinutes * 60 * 1000),
-        )
-
-        res.redirect(`/calendar/${utilities.getStartMonth()}`)
-      } else {
-        logError(req.url, '2FA failure')
-        res.render('pages/two-factor-auth', { authError: true, csrfToken: res.locals.csrfToken })
-      }
-    }),
-  )
-
-  const postLogin = asyncMiddleware(async (req, res) => {
-    // if maintenance start/end dates exist then dcheck whether to display maintenance page
-    // otherwise just ignore the following, it will become effective as soon as those environment
-    // variables are created.  13DEC19.
+  const postLogin = (req, res) => {
```

Comment:
Why did you move the function. Its obfuscating the actual changes you've made.

Reply:
Linting I think when you take something off (await?) you have to declare the function before you use it.